google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -236,6 +233,39 @@ public static ClientContext createClientContext(EnhancedBigtableStubSettings set
     return ClientContext.create(builder.build());
   }
 
+  private static void setInterceptors(
+      InstantiatingGrpcChannelProvider.Builder transportProvider,
+      EnhancedBigtableStubSettings.Builder settings) {
+    Set<ConnectionErrorCountInterceptor> interceptors =
+        setupConnectionErrorCountInterceptors(settings);
+    if (transportProvider != null) {
```

Comment:
Let's have this method return a list of interceptors.

```
ImmutableList.<ClientInterceptor>Builder builder = ImmutableList.builder()
    .add(new  ConnectionErrorCountInterceptor(interceptors))
if (settings.getEnableRoutingCookie()) {
    builder.add(new CookiesInterceptor());
}

return builder.build();

```

And on line 208 you can do 
```
if (transportProvider instanceof InstantiatingGrpcTransportProvider) {
   ((InstantiatingGrpcTransportProvider) transportProvider).setInterceptor(() -> ...)
}
```